java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,691 @@
+        } else {
+          currentBlock = createUnconditionalJump(s, body);
+        }
+        stepBlock.successors.add(currentBlock);
+        // process init
+        currentBlock = createBlock(currentBlock);
+        for (StatementTree init : Lists.reverse(s.initializer())) {
+          build(init);
+        }
+        break;
+      }
+      case TRY_STATEMENT: {
+        //FIXME only path with no failure constructed for now, (not taking try with resources into consideration).
+        TryStatementTree tryStatementTree = (TryStatementTree) tree;
+        currentBlock = createBlock(currentBlock);
+        BlockTree finallyBlock = tryStatementTree.finallyBlock();
+        if(finallyBlock != null) {
+          build(finallyBlock);
+        }
+        currentBlock = createBlock(currentBlock);
+        build(tryStatementTree.block());
+        break;
+      }
+      case THROW_STATEMENT: {
+        //FIXME this won't work if it is intended to be caught by a try statement.
+        ThrowStatementTree throwStatementTree = (ThrowStatementTree) tree;
+        currentBlock = createUnconditionalJump(throwStatementTree, exitBlock);
+        build(throwStatementTree.expression());
+        break;
+      }
+      case SYNCHRONIZED_STATEMENT: {
+        SynchronizedStatementTree sst = (SynchronizedStatementTree) tree;
+        //Naively build synchronized statement.
+        build(sst.block());
+        build(sst.expression());
+        break;
+      }
+      case POSTFIX_INCREMENT:
+      case POSTFIX_DECREMENT:
+      case PREFIX_INCREMENT:
+      case PREFIX_DECREMENT:
+      case UNARY_MINUS:
+      case UNARY_PLUS:
+      case BITWISE_COMPLEMENT:
+      case LOGICAL_COMPLEMENT:
+        UnaryExpressionTree e = (UnaryExpressionTree) tree;
+        currentBlock.elements.add(e);
+        build(e.expression());
+        break;
+      case PARENTHESIZED_EXPRESSION:
+        build(((ParenthesizedTree) tree).expression());
+        break;
+      case ARRAY_ACCESS_EXPRESSION: {
+        ArrayAccessExpressionTree aaet = (ArrayAccessExpressionTree) tree;
+        currentBlock.elements.add(aaet);
+        build(aaet.expression());
+        build(aaet.dimension());
+        break;
+      }
+      case ARRAY_DIMENSION: {
+        ArrayDimensionTree arrayDimensionTree = (ArrayDimensionTree) tree;
+        if(arrayDimensionTree.expression() != null) {
+          build(arrayDimensionTree.expression());
+        }
+        break;
+      }
+      case NEW_CLASS: {
+        NewClassTree newClassTree = (NewClassTree) tree;
+        currentBlock.elements.add(newClassTree);
+        if(newClassTree.classBody() != null) {
+          build(newClassTree.classBody());
+        }
+        build(newClassTree.identifier());
+        if(newClassTree.enclosingExpression() != null) {
+          build(newClassTree.enclosingExpression());
+        }
+        build(Lists.reverse(newClassTree.arguments()));
+      }
+      break;
+      case METHOD_REFERENCE:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)